ggshield/cmd/secret/ignore.py

```diff
@@ -7,41 +7,69 @@
 from ggshield.core.cache import Cache
 from ggshield.core.config import Config
 from ggshield.core.text_utils import pluralize
+from ggshield.core.types import IgnoredMatch
 
 
 @click.command()
+@click.argument(
+    "hash",
+    nargs=1,
+    type=str,
+    required=False,
+    metavar="HASH",
+)
 @click.option(
     "--last-found",
     is_flag=True,
     help="Ignore secrets found in the last `ggshield secret scan` run.",
 )
+@click.option(
+    "--name",
+    type=str,
+    help="Name of the secret to ignore.",
+)
 @add_common_options()
 @click.pass_context
 def ignore_cmd(
     ctx: click.Context,
+    hash: str,
+    name: str,
     last_found: bool,
     **kwargs: Any,
 ) -> None:
     """
     Ignore some secrets.
 
-    The `secret ignore` command instructs ggshield to ignore secrets it finds during a scan.
+    The `secret ignore` command instructs ggshield to ignore secrets.
 
-    This command needs to be used with an option to determine which secrets it should ignore.
-    For now, it only handles the `--last-found` option, which ignores all secrets found during the last scan.
+    Option `--name` allows to specify the name of the secret to ignore.
+
+    Option `--last-found` ignores all secrets found during the last scan.
 
     The command adds the ignored secrets to the `secrets.ignored-matches` section of your local
     configuration file. If no local configuration file is found, a `.gitguardian.yaml` file is created.
     """
+
+    ctx_obj = ContextObj.get(ctx)
+    config = ctx_obj.config
+    path = config.config_path
+
     if last_found:
-        ctx_obj = ContextObj.get(ctx)
-        config = ctx_obj.config
+        if hash or name:
+            raise click.UsageError(
+                "Option `--last-found` cannot be used with `HASH` or `--name`."
+            )
         nb = ignore_last_found(config, ctx_obj.cache)
-        path = config.config_path
-        secrets_word = pluralize("secret", nb)
-        click.echo(
-            f"Added {nb} {secrets_word} to the `secret.ignored-matches` section of {path}."
-        )
+    else:
+        match = IgnoredMatch(name=name if name else "", match=hash)
```

Comment:
I think we should make `--name` mandatory to encourage the good practice of naming secrets. Someone who does not want to do so can still use `--name ''`, but at least it's not the path of least resistance.

Reply:
I discussed this with Mathieu, and to summarize, the key point is that the `--name` option is only really helpful when well used. Forcing users to provide a name could lead to frustration, as they may input poorly chosen names just to bypass the requirement. In the end, these poorly named secrets would be as useless as no name.

Reply:
@mathieubellon ggshield is, on purpose, a source of frustration 😅. It gets in the way of users to prevent bad practices.

If the option is not mandatory I suspect it will just never be used because users won't even be aware of it.

Reply:
I would really like to enforce an option that will necessarily provide value (and not maybe provide value) and [our own example](https://github.com/GitGuardian/ggshield/blob/main/.gitguardian.yaml) shows that it is not the case.

If users need it, they will find it (like --ignore-known-secret)

I am also totally for finding the right solution to fix this problem (display secret in clear so it can be found?)

> ggshield is, on purpose, a source of frustration 😅

In its mission maybe, in its usage I would think (and love) not. I think our goal should be to provide the best user experience so it is actually used (and thus enforces good practice). It is the difference between Prettier and Go format for example IMO

Reply:
For example that would be better if ggshield actually automatically updates the yaml with useful informations (line number, file/commit, etc ..), maybe from the .cache_ggshield file ?

Reply:
> [our own example](https://github.com/GitGuardian/ggshield/blob/main/.gitguardian.yaml) shows that it is not the case

Many `name` fields in our config are empty because the matches were migrated from the v1 config format, which allowed users to use the secret SHA directly, without any `name` field, see this commit which did the migration for more context https://github.com/GitGuardian/ggshield/commit/11e3f0253e2774151feafe7f822d19866fa0d157. It would have been very painful to figure out which secrets the SHA were referring to, so the names were left blank. With hindsights, it would have been better to add a comment in them instead. 

> For example that would be better if ggshield actually automatically updates the yaml with useful informations (line number, file/commit, etc ..), maybe from the .cache_ggshield file ?

That's an interesting idea, the problem is the cache is regularly cleared so it would not be 100% reliable. What could be done is: if `--name` is not passed, prompt the user for a name, with the default value populated from the cache if available. What do you think?

Reply:
> if --name is not passed, prompt the user for a name, with the default value populated from the cache if available. What do you think?

That mean also that in case of the cli used in a scripted context the name is mandatory
Ok anyway, let's go with this